### PR TITLE
New pattern: aws-fargate-opensearch

### LINKED ISSUE #823: New Pattern: aws-fargate-opensearch
Create a construct that links a Fargate Service to Amazon OpenSearch Service

Should create a VPC or accept an existing VPC
Should create a Fargate Service or accept an existing service
Should create a new domain or accept an existing domain
Should grant the service IAM permission to write to endpoint
Should set up environment variables in the service that identify the domain endpoint

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by mickychetta**
*Issue #823, if available:*

*Description of changes:*
- Created new construct `aws-fargate-opensearch`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

fixes #823

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: a34aa74932653ed107bb0777b4f88fc3c80dccf6
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=392ec3d5-677a-4383-b7f4-45848b18f05e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: e29750c3520f641ae0694b567d98e5a041255236
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5cc0e833-9bfd-47d1-81b7-745703b2712d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 1d7e35746822e00b00d4c6a9498a384a873d767c
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1551783e-79e6-4b79-8007-7da7e35b6d86%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: f3ea038d9ae62bbe701df8f21df572f381f78c2a
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=745b7425-f502-474e-8c91-3ea8a9a5a3f2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — DISMISSED** [review]
One comment for a change going forward. At this point we should add OpenSearch to DESIGN_GUIDELINES.md

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 5ecfbad075f762011b9581b7952697655441ab89
* Result: SUCCEEDED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3d307644-81ae-408b-ba85-7f7393662dcc%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-fargate-opensearch/lib/index.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    };
+
+    [this.openSearchDomain, this.openSearchRole] = defaults.buildOpenSearch(this, buildOpenSearchProps);
+
+    if (props.createCloudWatchAlarms === undefined || props.createCloudWatchAlarms) {
+      this.cloudWatchAlarms = defaults.buildOpenSearchCWAlarms(this);
+    }
+
+    // Add environment variables
+    const bucketArnEnvironmentVariableName = props.domainEndpointEnvironmentVariableName || 'DOMAIN_ENDPOINT';
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-fargate-opensearch/lib/index.ts:206]: pretty sure bucketArnEnvironmentVariableName is not the variable name you want here.

File: source/patterns/@aws-solutions-constructs/aws-fargate-opensearch/test/fargate-opensearch.test.ts [hunk 2]
```diff
@@ -0,0 +1,725 @@
...
+  expect(stack).toHaveResourceLike("AWS::OpenSearchService::Domain", {
+    DomainName: DOMAIN_NAME
+  });
+
+  expect(stack).toHaveResourceLike("AWS::Cognito::UserPoolDomain", {
+    Domain: COGNITO_DOMAIN_NAME
+  });
+});
+
+test('Check construct property', () => {
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-fargate-opensearch/test/fargate-opensearch.test.ts:66]: properties

**Review by biffgaut — COMMENTED** [review]
Very minor stuff

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 63818a65526c931a1bb42a1bf9044523b68c8be7
* Result: SUCCEEDED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a5b92d21-6a28-4806-b780-53508472d623%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 7182f98e84d3b7feda09ff9ea1d08ef2271805a4
* Result: SUCCEEDED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=06eaeec7-aaf0-4c01-9dbe-9f991e2d85ac%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — APPROVED** [review]

**Pull request merged**